cmd/fastly/main.go

```diff
@@ -39,11 +38,119 @@ func main() {
 		out            io.Writer = common.NewSyncWriter(os.Stdout)
 	)
 
+	// We have to manually handle the inclusion of the verbose flag here because
+	// Kingpin doesn't evaluate the provided arguments until app.Run which
+	// happens later in the file and yet we need to know if we should be printing
+	// output related to the application configuration file in this file.
+	var verboseOutput bool
+	for _, seg := range args {
+		if seg == "-v" || seg == "-verbose" || seg == "--verbose" {
+			verboseOutput = true
+		}
+	}
+
+	// Extract a subset of configuration options from the local application directory.
+	var file config.File
+	err := file.Read(config.FilePath)
+	if err != nil {
+		if verboseOutput {
+			text.Output(out, `
+				We were unable to locate a local configuration file required to use the CLI.
+				We will create that file for you now.
+			`)
+			text.Break(out)
+		}
+
+		err := file.Load(config.RemoteEndpoint, httpClient)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1) // TODO: offer a clearer remediation step
+		}
+	}
+
+	// When the local configuration file is stale we'll need to acquire the
+	// latest version and write that back to disk. To ensure the CLI program
+	// doesn't complete before the write has finished, we block via a channel.
+	waitForWrite := make(chan bool)
+	wait := false
+
+	// Validate if configuration is older than its TTL
+	if check.Stale(file.CLI.LastChecked, file.CLI.TTL) {
+		if verboseOutput {
+			text.Warning(out, `
+Your local application configuration is out-of-date.
```

Comment:
This could be misleading – "local application" makes it sound like it could be an issue with their C@E project. I would maybe explain that this is the "CLI configuration".

Also rather than out of date... "the Fastly CLI configuration has recently changed, we'll refresh your local copy"? Sounds less like a problem.

Reply:
Snap!  I had the same comment - though we can't say for sure that it's changed, only that the current version is out of date.  I made a suggestion in my review.